### PR TITLE
UI support for Spark plugin types

### PR DESCRIPTION
**Pull request by elfenheart**

### CODE REVIEW

File: cdap-ui/app/features/hydratorplusplus/services/create/stores/left-panel-store.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ class HydratorPlusPlusLeftPanelStore {
   }
 
   getPlugins(extension) {
```
Comment (anwar6953) [inline cdap-ui/app/features/hydratorplusplus/services/create/stores/left-panel-store.js]: nit - there is a usage of this in another file that redundantly does `|| []` now. That can be removed.

**Comment by anwar6953** [comment]
Can you post a screenshot or something of how this looks in the UI?
Ideally, the SparkSink should be in the sink section of the plugins, and not its own section.

File: cdap-ui/app/features/hydratorplusplus/services/create/stores/left-panel-store.js (hunk 1 above)
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/services/create/stores/left-panel-store.js]: updated

**Comment by elfenheart** [comment]
@anwar6953 

screen shot 2016-03-28 at 5 00 01 pm

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**